src/api/docgen/transpile/python.ts

```diff
@@ -0,0 +1,323 @@
+          const parameter = propertyToParameter(callable, property);
+          parameters.push(parameter);
+        }
+      }
+    }
+
+    const name = toSnakeCase(callable.name);
+    const inputs = parameters.map((p) => formatInput(this.parameter(p)));
+
+    return {
+      name,
+      parentType: type,
+      import: formatImport(type),
+      parameters,
+      signature: formatSignature(name, inputs),
+      invocation: formatInvocation(
+        type,
+        inputs,
+        callable.kind === reflect.MemberKind.Initializer ? undefined : name
+      ),
+    };
+  }
+
+  public interface(
+    iface: reflect.InterfaceType
+  ): transpile.TranspiledInterface {
+    return {
+      name: iface.name,
+      type: this.type(iface),
+    };
+  }
+
+  public type(type: reflect.Type): transpile.TranspiledType {
+    const module = this.moduleFqn(type.fqn);
+
+    const fqn = [module];
+
+    if (type.namespace) {
+      fqn.push(type.namespace);
+    }
+
+    fqn.push(type.name);
+
+    return { fqn: fqn.join("."), moduleFqn: module };
+  }
+
+  private moduleFqn(fqn: string): string {
+    const type = this.ts.findFqn(fqn);
+    const submodule = this.findSubmodule(type);
+
+    const targets = submodule ? submodule.targets : type.assembly.targets;
+
+    if (!targets) {
+      throw new Error(`Unable to find 'targets' for fqn ${type.fqn}`);
+    }
+
+    if (!targets.python) {
+      throw new Error(`Python is not a supported target for fqn ${type.fqn}`);
+    }
+
+    return targets.python.module;
+  }
+
+  private optionalityCompare(
+    p1: reflect.Parameter,
+    p2: reflect.Parameter
+  ): number {
+    if (!p1.optional && p1.optional) {
+      return -1;
+    }
+    if (!p2.optional && p1.optional) {
+      return 1;
+    }
+    return 0;
+  }
+
+  private isStruct(p: reflect.Parameter): boolean {
+    return p.type.fqn ? p.system.findFqn(p.type.fqn).isDataType() : false;
+  }
+
+  private findSubmodule(type: reflect.Type): reflect.Submodule | undefined {
+    if (!type.namespace) {
+      return undefined;
+    }
+
+    // if the type is in a submodule, the submodule name is the first
+    // part of the namespace. we construct the full submodule fqn and seach for it.
+    const submoduleFqn = `${type.assembly.name}.${
+      type.namespace.split(".")[0]
+    }`;
+    const submodules = type.assembly.submodules.filter(
+      (s) => s.fqn === submoduleFqn
+    );
+
+    if (submodules.length > 1) {
+      // can never happen, but the array data structure forces this handling.
+      throw new Error(`Found multiple submodulues with fqn ${submoduleFqn}`);
+    }
+
+    if (submodules.length === 0) {
+      return undefined;
+    }
+
+    // type is inside this submodule.
+    return submodules[0];
+  }
+
+  private typing(type: "List" | "Mapping" | "Any" | "Union"): string {
+    return `typing.${type}`;
+  }
+
+  private builtins(type: "bool" | "str") {
+    return `builtins.${type}`;
```

Comment:
I understand this is how this is expressed in mypy, but do we really need the `builtins.` prefix everywhere?

Generally speaking: I am not sure that the mypy representation is the most friendly way to present the reference in python.

Reply:
I don't actually think its related to `mypy` here...i'm not sure why the code generator uses `builtins.str` and not just `str`. I'll ping @RomainMuller about it and in the meantime change to just use `bool` and `str`.

I wonder also if we should remove the type hints from the code snippet, making it much more easy on the eyes, and leaving types only in the dedicated parameter section.

Reply:
> I wonder also if we should remove the type hints from the code snippet, making it much more easy on the eyes, and leaving types only in the dedicated parameter section.
 
Either omit completely or format in a more human-friendly way